packages/trie/src/trie/trie.ts

```diff
@@ -116,6 +145,10 @@ export class Trie {
    * @returns A Promise that resolves once value is stored.
    */
   async put(key: Buffer, value: Buffer): Promise<void> {
+    if (this._persistRoot) {
+      ok(!key.equals(ROOT_DB_KEY), `Attempted to set '__root__' key but it is not allowed.`)
```

Comment:
Let's directly `throw new Error` here instead of using assert utilities. We recently removed all references to `assert` across the monorepo and don't want to reintroduce them.